File: Python/Sorting/selection-sort.py
def selectionSort(nums):
    n = len(nums)
    for i in range(n):
        min_index = i
        for j in range(i+1, n):
            if nums[min_index] > nums[j]:
                min_index = j
        nums[i], nums[min_index] = nums[min_index], nums[i]
    return nums

nums = []
n = int(input("Enter n: "))
for i in range(n):
    element = int(input("Enter element {}: ".format(i+1)))
    nums.append(element)

print()
print("Current array:", nums)
selectionSort(nums)
print("Sorted array:", nums)

File: C/Sorting/selection-sort.c
#include <stdio.h>

void selectionSort(int arr[], int n);
void display(int arr[], int n);

int main() {

    int arr[] = {4, 7, 1, 2, 8};
    selectionSort(arr, 5);

}

void selectionSort(int arr[], int n) {

    int i, j, min_index;

    for (i = 0; i < n-1; ++i) {
        min_index = i;
        for (j = i+1; j < n; ++j) {
            if (arr[j] < arr[min_index]) {
                min_index = j;
            }
        }
        int temp = arr[i];
        arr[i] = arr[min_index];
        arr[min_index] = temp;
        display(arr, n);
    }

}

void display(int arr[], int n) {

    printf("Current array: ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", arr[i]);
    }
    printf("\n");

}